src/libostree/ostree-repo.c

```diff
@@ -5290,6 +5290,47 @@ _ostree_repo_allocate_tmpdir (int tmpdir_dfd,
   return ret;
 }
 
+/**
+ * ostree_repo_delete_compat_signature:
+ * @self: Self
+ * @commit_checksum: SHA256 of given commit to delete signature from
+ * @signature: Signature data
+ * @cancellable: A #GCancellable
+ * @error: a #GError
+ *
+ * Delete a GPG signature from a commit's compat signature file.
+ */
+gboolean
+ostree_repo_delete_compat_signature (OstreeRepo     *self,
+                                     const gchar    *commit_checksum,
+                                     GBytes         *signature,
+                                     GCancellable   *cancellable,
+                                     GError        **error)
+{
+  g_autoptr(GBytes) compat_sig = NULL;
+
+  if (!_ostree_repo_read_commit_compat_signature (self, commit_checksum,
+                                                  &compat_sig, cancellable,
+                                                  error))
+      return FALSE;
+
+  if (g_bytes_equal (compat_sig, signature))
```

Comment:
`_ostree_repo_read_commit_compat_signature()` will return `TRUE` and a `NULL` `compat_sig` when it's not found -- so I think this function should check for that and return `TRUE` directly when `compat_sig == NULL`